packages/permission-controller/src/PermissionController.ts

```diff
@@ -1885,47 +1936,115 @@ export class PermissionController<
       origin,
     };
 
-    const permissionsRequest = {
+    const permissionsRequest: PermissionsRequest = {
       metadata,
       permissions: requestedPermissions,
     };
 
     const approvedRequest = await this.requestUserApproval(permissionsRequest);
-    const { permissions: approvedPermissions, ...requestData } =
-      approvedRequest;
+    return await this.#handleApprovedPermissions({
+      subject,
+      metadata,
+      preserveExistingPermissions,
+      approvedRequest,
+    });
+  }
 
-    const sideEffects = this.getSideEffects(approvedPermissions);
+  /**
+   * Initiates an incremental permission request that prompts for user approval.
+   * Incremental permission requests allow the caller to replace existing and/or
+   * add brand new permissions and caveats for the specified subject.
+   *
+   * Incremental permission request are merged with the subject's existing permissions
+   * through a right-biased union, where the incremental permission are the right-hand
+   * side of the merger. If both sides of the merger specify the same caveats for a
+   * given permission, the caveats are merged using their specification's caveat merger
+   * property.
+   *
+   * Either this or {@link PermissionController.requestPermissions} should
+   * always be used to grant additional permissions to a subject, unless user
+   * approval has been obtained through some other means.
```

Comment:
I think we are gonna need a way to both grant and revoke incremental permissions outside of this function as well. For cases where user approval has already been obtained.

Reply:
Revoke will be done in a follow-up per #4236, but I will try to add `grantIncrementalPermissions` in this PR.

Reply:
Added: [5908e92](https://github.com/MetaMask/core/pull/4222/commits/5908e929d4caf6a69bf8159d95536e89b876f410)